app/code/Magento/Inventory/Ui/Component/MassAction/Filter.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Ui\Component\MassAction;
+
+use Magento\Ui\Component\MassAction\Filter as BaseFilter;
+
+/**
+ * @author  naydav <valeriy.nayda@gmail.com>
+ * @todo remove this class after code from Engine\MagentoFix\Ui\Component\MassAction\Filter was deploy to production
+ */
+class Filter
+{
+    /**
+     * @var BaseFilter
+     */
+    private $filter;
+
+    /**
+     * @param BaseFilter $filter
+     */
+    public function __construct(
+        BaseFilter $filter
+    ) {
+        $this->filter = $filter;
+    }
+
+    /**
+     * @return array
+     */
+    public function getIds()
+    {
+        $this->filter->applySelectionOnTargetProvider();
+        $component    = $this->filter->getComponent();
+        $dataProvider = $component->getContext()->getDataProvider();
+        $searchResult = $dataProvider->getSearchResult();
+        $ids          = [];
+        foreach ($searchResult->getItems() as $item) {
+            $ids[] = $item->getId();
+        }
```

Comment:
I prefer to use next syntax : 
```php
array_map(function(\Magento\Framework\Api\Search\DocumentInterface $item) {
    return $item->getId();
}, $searchResult->getItems());
```
Doing that way you can ensure the type validation of the array returned